packages/react-components/src/components/Date/SimpleDate.jsx

```diff
@@ -41,6 +42,15 @@ export const SimpleDate = ({
   const inputId = useRef(uniqueId('date-input-'));
   const { day, month, year } = date;
 
+  const [lang, setLang] = useState();
+
+  useEffect(() => {
+    i18next.on('languageChanged', lang => {
+      console.log('language changed SimpleDate');
+      setLang(lang);
+    });
+  }, []);
```

Comment:
This effect hook is [basically like the `componentDidMount()` lifecycle event](https://reactjs.org/docs/hooks-reference.html#conditionally-firing-an-effect):

>  While passing `[]` as the second argument is closer to the familiar `componentDidMount` and `componentWillUnmount` mental model ...

We're doing this because `i18next` changing the language it's using internally as a result of the mutation observer doesn't trigger a re-render. Teams who use the component library _could_ force a re-render themselves and the output might be what they expect, but in the testing I did in storybook it's best if we have our translation-enhanced components force themselves to re-render when a language change is detected. With this approach the translation happens automatically _as long as_ the dynamic language switching also updates the `lang` attribute in `<main>`.

The state variable isn't really used other than to trigger the re-render so there's probably a better way of triggering a re-render, even if it's just naming it `_lang`.